src/ocean/meta/codegen/CTFE.d

```diff
@@ -119,3 +120,52 @@ unittest
     assert (stripQualifiedPrefix("aabb") == "aabb");
     static assert (stripQualifiedPrefix("aabb") == "aabb");
 }
+
+
+/******************************************************************************
+
+    Converts a snake_case name to a PascalCase.
+    This should be used only at CTFE since is not memory efficient.
+
+    Params:
+        name = the event name
+
+    Returns:
+        The name in camel case
+
+******************************************************************************/
+
+string snakeToPascal ( string name )
+{
+    string result;
```

Comment:
You could add a static assert that this is CTFE (I think there is something like `_ctfe` or something you can check)

Reply:
why you would forbid usage of this function at runtime?

Reply:
I'm just going off your comment in the documentation that says it should only be used at compile time.

Reply:
yeah... but this is why I wrote should.. if someone wants to use this at run time, they should assume the gc allocations

Reply:
Maybe instead of documenting it should be used only at compile time you could indicate that it is allocating memory.

Reply:
@matthias-wende-sociomantic i am not sure that I understand that phrase.

Reply:
He means that instead of saying "This should be used only at CTFE since is not memory efficient.", you could say something like "This function is designed for use at compile time and is thus not careful about avoiding allocations.".

Reply:
ah cool. I updated the comment.